### DISCUSSION TITLE
Doubt in dataset annotation for training rec model.

### DISCUSSION THREAD

**Discussion by kishore-work-hard** [PaddlePaddle/PaddleOCR#13418] [Q&A] [answered]
my image contains two lines. and i have marked those two lines under two different bounding boxes - 
![115_jpg rf a5d5827fbb8f30b135d1951ceb48157b_car_0](https://github.com/user-attachments/assets/80bcdbb0-12f9-4494-9ad4-b2ccd5c133b1)

**my question is** - 
while training for recognition model. should i crop those two lines and then save those two images in a folder and then for each image should i annotate it like

`filepath tabspace text`
?

in the above image the label.txt file would be - 
```
path tab KLO7CJ
path tab 8999
```

?

or should i not crop this image into two and simply mark the label as 
`path tab KLO7CJ8999`

which is the best approach for best result? these are number plates by the way? 

and where can i find the  ENGLISH configuration yml file for training text detection model ? all i see is Chinese in `config/dec` folder

**Comment by kishore-work-hard** [comment]
@GreatV

**Reply by GreatV** [reply]
The CRNN model can't recognize multiple lines of text, so you need to split the content into two lines during the detection phase.

**Reply by kishore-work-hard** [reply]
> The CRNN model can't recognize multiple lines of text, so you need to split the content into two lines during the detection phase.

No I was asking about recognition.
 Do I need to split the image into two lines during recognition training?  

Like which is correct 

**After splitting the image:**
```
path tab KLO7CJ
path tab 8999
```
Or 

**Without splitting the image:**
`path tab KLO7CJ8999`

@GreatV ?
When I meant splitting... I will manually crop the text detection bounding boxes and save them as two seperate images and then train the RECOGNITION MODEL. 
Which is the best approach. Idk. Please guide me.

**Reply by GreatV** [reply]
The CRNN model can't recognize multiple lines of text，you need split the image.

**Reply by kishore-work-hard** [reply]
> The CRNN model can't recognize multiple lines of text，you need split the image.

**Doubt**
While training a recognition model 
During the training process does the algorithm do text detection ?

**Last question -** 
I have trained my text DETECTION model successfully. 

Do I need to give this custom text detection model as an input for recognition training 
while training for text RECOGNITION?

**Reply by kishore-work-hard** [reply]
@GreatV 

> > The CRNN model can't recognize multiple lines of text，you need split the image.
> 
> **Doubt** While training a recognition model During the training process does the algorithm do text detection ?
> 
> **Last question -** I have trained my text DETECTION model successfully.
> 
> Do I need to give this custom text detection model as an input for recognition training while training for text RECOGNITION?

@GreatV

**Reply by GreatV** [reply, marked as answer]
hi @kishore-work-hard,

When training a text recognition model using PaddleOCR, the text detection step is typically separate from the text recognition step. Here’s a detailed explanation:

### Text Detection and Recognition Workflow

1. **Text Detection:**
   - This step involves identifying and localizing text regions within an image. The output is generally bounding boxes around the detected text areas.
   - You’ve already successfully trained your text detection model, which means you have a model capable of identifying where the text is located in images.

2. **Text Recognition:**
   - This step involves recognizing and transcribing the text within the detected text regions.
   - For training a text recognition model, you need cropped images of the text regions and their corresponding labels (the actual text content).
   - **CTC (Connectionist Temporal Classification) Decoding:** Many text recognition models use CTC decoding to handle sequences where the alignment between the input (image) and output (text) is not known. CTC allows the model to predict a probability distribution over all possible alignments, making it suitable for recognizing text in images with varying lengths and styles.

### Does the Recognition Training Process Include Text Detection?

No, the text recognition training process does not inherently include text detection. During recognition model training, the input is typically already pre-cropped text images, not the full images where the text detection needs to occur.

### Why Text Recognition Models Cannot Recognize Multiple Lines of Text

Text recognition models are generally designed to recognize a single line of text at a time. Here’s why:

- **Training Data:** The recognition model is trained on images that contain single lines of text, making it specialized for this type of input. Multiple lines of text would require a different approach and training data.
- **Model Architecture:** The architecture of text recognition models is optimized for single-line text sequences. Handling multiple lines would necessitate changes in how the model processes input images.
- **CTC Decoding:** CTC decoding, commonly used in text recognition, is designed to handle sequence alignment for single lines of text. Extending this to multiple lines would complicate the decoding process and reduce accuracy.

### Multi-line Text Recognition and End-to-End Models

Recognizing multiple lines of text typically requires an end-to-end model that integrates both text detection and recognition. Here’s why:

- **Integrated Approach:** An end-to-end model processes the entire image, detecting and recognizing text in a single pass. This approach can handle multiple lines of text more effectively than separate detection and recognition models.
- **Complex Layouts:** End-to-end models can manage complex text layouts and multiple lines of text, which is challenging for traditional text recognition models designed for single-line text.
- **Training Process:** End-to-end models are trained on datasets containing images with multiple lines of text, learning to detect and recognize text regions within a single framework.

### Using Custom Text Detection Model for Recognition Training

- **Preprocessing Step:** Before training the text recognition model, you should use your custom text detection model to preprocess your dataset. This involves using the detection model to detect text regions in the images and then cropping these regions to create the training data for the recognition model.
- **CTC Decoding in Training:** During the training of the text recognition model, CTC decoding is applied to handle the variable-length sequences and align the predicted text with the ground truth labels effectively.
- **Pipeline for Inference:** Once both models are trained, the typical inference pipeline involves first using the text detection model to detect text regions in an image, and then passing these detected regions to the text recognition model for transcription.

### Summary

1. **Training:** Train the text detection and text recognition models separately.
2. **Data Preparation for Recognition:** Use the text detection model to preprocess images by cropping text regions, and then use these cropped regions as input for training the text recognition model. CTC decoding will be used during the recognition training to handle sequence alignment.
3. **Inference Pipeline:** Use the detection model to detect text areas in images and then use the recognition model with CTC decoding to transcribe the detected text.

For multi-line text recognition, consider using an end-to-end model that integrates text detection and recognition within a single framework. This approach allows for handling complex text layouts and multiple lines of text more effectively than separate models designed for single-line text recognition.
